circleguard/gui.py

```diff
@@ -1002,7 +1277,8 @@ def __init__(self):
         # we want widgets to fill from top down,
         # being vertically centered looks weird
         self.layout.setAlignment(Qt.AlignTop)
-        self.info_label = QLabel("After running Comparisons, this tab will fill up with results")
+        self.info_label = QLabel("After running checks, this tab will fill up "
+                                 "with replays that can be played back.")
```

Comment:
why the line break? If you want to follow some pep rule or something please apply it to everything

Reply:
It's to avoid going over 80 characters in a line.

<img width="644" alt="Screen Shot 2020-01-27 at 9 32 55 PM" src="https://user-images.githubusercontent.com/31628143/73230865-99179180-414c-11ea-9da4-7343a9d745b6.png">

My linter doesn't yell at me for it, I just try and follow it where possible. I don't think it's good to follow it religiously (basically all of settings.py goes over, which is fine - that particular file needs a lot of horizontal space for text), and reformatting the entire codebase to follow by hand would be tedious. It should be done at some point, but I've been holding off.

Reply:
>I just try and follow it where possible.

This... I... I understand what you mean but we can't just do however we want it. The outcome is code that is horrible to read. I would much prefer following the rule or not.

>  and reformatting the entire codebase to follow by hand would be tedious.

Every IDE can do this automatically, you just have to manually check it before commiting.

> It should be done at some point, but I've been holding off.

I would prefer doing it all at once over just randomly changing some lines

Reply:
IDEs can do it automatically - that doesn't mean they can do it well. Sometimes splitting a comment at a certain word could cause confusion or ambiguity ("did the writer of the comment mean for the next line to be a continuation of the comment on the previous line, or a new sentence?"), which all needs to be dealt with manually. The effort to reviewing that automated formatting is nontrivial.

And we can in fact to it however we want - I don't think that code is horrible to read at all, even when some sections ignore the 80 character limit and some don't. Take cacher.py in circlecore, which (as of the time of writing) is about half and half for following the 80 character limit. I think it looks better than if none of the file followed the 80 character limit.

Reply:
In `catcher.py` there is [a single line](https://github.com/circleguard/circlecore/blob/master/circleguard/cacher.py#L78) above 150 chars, which doesn't look any worse/better when you split it to a new line. Therefore I would set the line length limit to 150 chars. There isn't a single line that is being split that is shorter than 150 chars.

You just changed the limit from a 80 char limit to a 150 char limit. I don't see your point of "sometimes it looks better, sometimes it doesn't".

> I don't think that code is horrible to read at all

Sorry, let me rephrase. While reading the code, the thing I would have in my mind would be "Why is this line being split for absolutely no reason?"

Reply:
> You just changed the limit from a 80 char limit to a 150 char limit.

not what's happening at all! `cacher.py` should be rewritten to use 80 character line limits. I didn't mean to hold it up as an idea file, more of an example where half of the file follows style guidelines and the other doesn't, and looks better than if the entire file didn't follow the guidelines.

>  "sometimes it looks better, sometimes it doesn't".

if you're referring to why I think it's appropriate for `settings.py` to ignore this limit, that's because it has *so many long strings*. The vertical length of the file would be massive if we limited those all to 80 characters. It just doesn't make sense to apply the style guidelines there when there's a block of long strings. If there were only one long string, it would make sense to split it across multiple lines.

The bottom half of `settings.py` still follows (going a couple characters over in some places, but close enough) the 80 character guideline. I think the file as a whole looks better than if the bottom half ignored this guideline.

Reply:
> more of an example where half of the file follows style guidelines and the other doesn't

But that isn't the case! You didn't split a single line that is under 150 chars long, even though they are over 80 chars. You obviously followed the 150 char limit rule. You did not follow the 80 char rule sometimes. The freaking lines you split are still longer than 80 chars ![](https://i.fiery.me/l2y5m.png)

> if you're referring to why I think it's appropriate for settings.py to ignore this limit

I fully agree with ignoring the limit for settings.py, the long lines are data (In my eyes at least), not code.

> The bottom half of settings.py still follows (going a couple characters over in some places, but close enough) the 80 character guideline. I think the file as a whole looks better than if the bottom half ignored this guideline.

Again. There is not a single line that is being split in the bottom section. All code is shorter than 80 chars besides one comment (which can be ignore imo, since it isn't code). If you disabled the 80 char limit, and format the file so every line is as long as possible, _`settings.py` will not change by a single bit._

Reply:
> You didn't split a single line that is under 150 chars long,

I split comments; I consider those just as valid as code for following 80 characters. You're right that I didn't split any line of code that was over 80 lines, but we're getting off track from the original debate. I'm not trying to say that `cacher.py` has been intentionally modified to fit within 80 characters, because in this case it hasn't. All I'm saying is that files look better when they *sometimes* follow the 80 character limit than when they don't at all.

Reply:
In my eyes code and comments are 2 different things, but as you mentioned this is getting off track. We will keep it how it is.